api/src/controllers/login.js

```diff
@@ -355,6 +431,60 @@ module.exports = {
       });
   },
 
+  getPasswordReset: (req, res, next) => {
+    return renderPasswordReset(req)
+      .then(body => {
+        res.setHeader(
+          'Link',
+          '</login/style.css>; rel=preload; as=style, '
+              + '</login/auth-utils.js>; rel=preload; as=script, '
+              + '</login/password-reset.js>; rel=preload; as=script'
+        );
+        res.send(body);
+      })
+      .catch(next);
+  },
+  resetPassword: async (req, res) => {
+    const limited = await rateLimitService.isLimited(req);
+    if (limited) {
+      return serverUtils.rateLimited(req, res);
+    }
+
+    try {
+      const validation = validatePassword(req.body.password, req.body.confirmPassword);
+      if (!validation.isValid) {
+        return res.status(400).json({
+          error: `password-${validation.error}`,
+          params: validation.params,
+        });
+      }
+      const userCtx = await auth.getUserCtx(req);
+      const user = await db.users.get(`org.couchdb.user:${userCtx.name}`);
+      user.password = req.body.password;
+      user.password_change_required = false;
+
+      await db.users.put(user);
+
+      cookie.clearCookie(res, 'login');
+      cookie.clearCookie(res, 'AuthSession');
```

Comment:
This line implies the AuthSession is already created at this point based on the old user/pass. When we last talked about it I think we agreed you cannot create a session using the password that requires reset, because it would be trivial to use that session and bypass the password reset.

Reply:
I get it now, the current setCookie creates the AuthSession so I need to should refactor it to only set AuthCookie for users who do not need password reset.